File: 2023/1/part2.py
f = open("input.txt", "r")
input = f.readlines()
f.close()

sum = 0

for line in input:
    nums = []
    print(line)
    line = line.replace("one", "one1one")
    line = line.replace("two", "two2two")
    line = line.replace("three", "three3three")
    line = line.replace("four", "four4four")
    line = line.replace("five", "five5five")
    line = line.replace("six", "six6six")
    line = line.replace("seven", "seven7seven")
    line = line.replace("eight", "eight8eight")
    line = line.replace("nine", "nine9nine")
    for char in line:
        try:
            nums.append(int(char))
        except Exception:
            pass
    sum += 10 * nums[0] + nums[-1]
    print(line)
    print(nums[0], nums[-1])

print(sum)

File: 2015/8/part2.cpp
#include <fstream>
#include <iostream>
#include <string>
#include <string_view>

int newLength(std::string_view s) {
  int length = 2;

  for (char c : s) {
    if (c == '\\' || c == '"') {
      length += 2;
    } else {
      length += 1;
    }
  }

  return length;
}

int main() {
  std::ifstream file;

  file.open("input.txt");

  std::string line = "";
  int sum = 0;

  while (std::getline(file, line)) {
    int codeLength = line.length();
    sum += newLength(line) - codeLength;
    std::cout << newLength(line);
    std::cout << " - ";
    std::cout << codeLength;
    std::cout << " = ";
    std::cout << newLength(line) - codeLength << "\n";
  }

  std::cout << sum;

  return 0;
}
